documentation/src/main/asciidoc/reference/coordination.asciidoc

```diff
@@ -0,0 +1,508 @@
+* leveraging Hibernate ORM's batch fetching:
+see link:{hibernateDocUrl}#configurations-database-fetch[the `batch_fetch_size` property]
+and link:{hibernateDocUrl}#fetching-batch[the `@BatchSize` annotation].
+* leveraging Hibernate ORM's link:{hibernateDocUrl}#caching[second-level cache],
+especially for immutable entities referenced from indexed entities
+(e.g. for reference data such as countries, cities, ...).
+[[coordination-none-indexing-guarantee]]
+Indexing is not guaranteed on commit, but only when the application thread returns::
+[[mapper-orm-indexing-automatic-concepts-transaction-commit]] When entity changes happen inside a transaction,
+indexes are not updated immediately, but only after the transaction is successfully committed.
+That way, if a transaction is rolled back, the indexes will be left in a state consistent with the database,
+discarding all the index changes that were planned during the transaction.
++
+However, if an error occurs in the backend while indexing,
+this behavior means that <<limitations-backend-indexing-error,index changes may be lost, leading to out-of-sync indexes>>.
+If this is a problem for you, you should consider switching to <<coordination-database-polling,another coordination strategy>>.
++
+[[mapper-orm-indexing-automatic-concepts-notransaction-flush]]
+NOTE: When entity changes happen outside of any transaction (not recommended),
+indexes are updated immediately upon session `flush()`.
+Without that flush, indexes will not be updated automatically.
+[[coordination-none-indexing-visibility]]
+Index changes may not be visible immediately::
+[[mapper-orm-indexing-automatic-concepts-nearrealtime]] By default, indexing will resume the application thread after index changes are committed to the indexes.
+This means index changes are safely stored to disk,
+but this does not mean a search query ran immediately after indexing will take the changes into account:
+when using the Elasticsearch backend in particular, changes may take some time to be visible from search queries.
++
+See <<mapper-orm-indexing-automatic-synchronization>> for details.
+[[coordination-none-indexing-association-consistency]]
+Associations must be updated on both sides::
+[[mapper-orm-indexing-automatic-concepts-session-consistency]] Hibernate ORM is able to handle asymmetric updates of associations,
+where only the owning side of association is updated and the other side is ignored.
+The entities in the session will be inconsistent for the duration of the session,
+but upon reloading they will be consistent once again,
+due to how entity loading works.
++
+This dodgy practice of asymmetric updates of associations
+can cause problems in applications in general,
+but also in Hibernate Search specifically,
+where it may lead to out-of-sync indexes.
+Thus, it must be avoided.
++
+[NOTE]
+====
+For example, let's assume an indexed entity `A` has an `@IndexedEmbedded` association `A.b`to entity `B`,
```

Comment:
there are a formatting issue, maybe we should add a space just after the term \`A.b\` here before the `to`